demos/common/cpp/pipelines/src/requests_pool.cpp

```diff
@@ -23,6 +23,13 @@ RequestsPool::RequestsPool(InferenceEngine::ExecutableNetwork& execNetwork, unsi
     }
 }
 
+RequestsPool::~RequestsPool() {
+    // Setting empty callback to free resources allocated for previously assigned lambdas
+    for (auto& pair : requests) {
+        pair.first->SetCompletionCallback([]{});
```

Comment:
`AsyncPipeline` is the one assigning callbacks. It makes it its responsibility to free them.

Reply:
Actually that approach still delays resource freeing (`internalModelData`, `metaData`). We could free them once callback finishes: https://github.com/openvinotoolkit/open_model_zoo/blob/27c311d2f9034d95ffde7180c9472ae635b8e8df/demos/security_barrier_camera_demo/cpp/main.cpp#L494

Reply:
agreed. Code resetting callback is moved to lambda.